identity_verification/src/verification_method/method.rs

```diff
@@ -245,3 +245,42 @@ impl KeyComparable for VerificationMethod {
     self.id()
   }
 }
+
+// Horrible workaround for a tracked serde issue https://github.com/serde-rs/serde/issues/2200. Serde doesn't "consume"
+// the input when deserializing flattened enums (MethodData in this case) cousing duplication of data (in this case
+// it ends up in the properties object). This workaround simply remove the duplication.
+#[derive(Deserialize)]
+struct _VerificationMethod {
+  #[serde(deserialize_with = "deserialize_id_with_fragment")]
+  pub(crate) id: DIDUrl,
+  pub(crate) controller: CoreDID,
+  #[serde(rename = "type")]
+  pub(crate) type_: MethodType,
+  #[serde(flatten)]
+  pub(crate) data: MethodData,
+  #[serde(flatten)]
+  pub(crate) properties: Object,
+}
+
+impl From<_VerificationMethod> for VerificationMethod {
+  fn from(value: _VerificationMethod) -> Self {
+    let _VerificationMethod {
+      id,
+      controller,
+      type_,
+      data,
+      mut properties,
+    } = value;
+    let data_json = serde_json::to_value(&data).unwrap();
+    let data_type = data_json.as_object().unwrap().into_iter().next().unwrap().0;
+    properties.remove(data_type);
```

Comment:
Hm, I guess we can just ignore this rant here, but couldn't bring myself to not complain about the `.unwrap()` here. :D

Could this be refactored into a match/if-let version that just removes the type (also adressed by its name instead of its relative position in the deserialized JSON) and does nothing if for whatever reasons the field couldn't be found?

Sorry for stirring up a (most probably useless) discussion but I've to be reaaally careful with `.unwrap()`s in library code. ^^

Reply:
Looks good to me. ^^